packages/react-composites/tests/browser/chat/app/identity.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { ChatClient } from '@azure/communication-chat';
+import { AzureCommunicationTokenCredential } from '@azure/communication-common';
+import { CommunicationIdentityClient, CommunicationUserToken } from '@azure/communication-identity';
+
+export const createUserAndThread = async (
+  resourceConnectionString: string,
+  topic: string,
+  displayNames: string[]
+): Promise<Array<any>> => {
+  const endpointUrl = new URL(resourceConnectionString.replace('endpoint=', '').split(';')[0]).toString();
+  const tokenClient = new CommunicationIdentityClient(resourceConnectionString);
+  const userAndTokens: CommunicationUserToken[] = [];
+  for (let i = 0; i < displayNames.length; i++) {
+    userAndTokens.push(await tokenClient.createUserAndToken(['chat']));
+  }
+
+  const chatClient = new ChatClient(endpointUrl, new AzureCommunicationTokenCredential(userAndTokens[0].token));
+  const threadId = (await chatClient.createChatThread({ topic: topic })).chatThread?.id ?? '';
+  await chatClient.getChatThreadClient(threadId).addParticipants({
+    participants: displayNames.map((displayName, i) => ({ id: userAndTokens[i].user, displayName: displayName }))
+  });
+
+  return displayNames.map((displayName, i) => ({
```

Comment:
How are return an Object to make it type-strong?

Or define the returned array/Tuple like [CommunicationUserIdentifier, string, string, string ,string]

Then you can spread this array like

createAzureCommunicationChatAdapter(...ret, displayName)

Reply:
Adding types. But can't spread it since I am using the 0th index of the result.